src/peft/tuners/lora/awq.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Optional
+
+import torch
+
+from peft.import_utils import is_auto_awq_available
+from peft.tuners.lora.layer import LoraLayer
+from peft.tuners.tuners_utils import BaseTunerLayer
+
+
+if is_auto_awq_available():
+    from awq.modules.linear import WQLinear_GEMM as AWQ_WQLinear_GEMM
+
+
+class WQLinear_GEMM(torch.nn.Module, LoraLayer):
```

Comment:
Not a fan of `_` in class names, mixing snake with camel case. I see that the original layer also has the `_`, but there isn't really a need for us to use the exact same name, is there (in fact, it could be more confusing)? I can see an argument for consistency with other PEFT layer names, but it's just ugly :D

Reply:
haha yeah makes sense, I agree it is ugly. OK I'll change it

Reply:
Done !